tools/hangar/__snapshots__/invalid.ts.snap

```diff
@@ -3009,11 +3009,11 @@ Duration <DURATION>"
 `;
 
 exports[`resource_captures.test.w 1`] = `
-"error: Cannot qualify access to a lifted object of type \\"Bucket\\" (see https://github.com/winglang/wing/issues/76 for more details)
-   --> ../../../examples/tests/invalid/resource_captures.test.w:13:13
+"error: Reference to lifted preflight object of type \\"Bucket\\" of unknown origin, can't qualify the lift (see https://github.com/winglang/wing/issues/76 for details)
+   --> ../../../examples/tests/invalid/resource_captures.test.w:15:5
    |
-13 |     let b = this.bucket;
-   |             ^^^^^^^^^^^ Cannot qualify access to a lifted object of type \\"Bucket\\" (see https://github.com/winglang/wing/issues/76 for more details)
+15 |     b.put(\\"hello\\", \\"world\\");
+   |     ^ Reference to lifted preflight object of type \\"Bucket\\" of unknown origin, can't qualify the lift (see https://github.com/winglang/wing/issues/76 for details)
```

Comment:
Previously qualification errors where reported when you had a preflight object referenced without any property:
```wing
let b = new cloud.Bucket();
test "test" {
  let x = b; // We're referencing b without any qualification so we used to report an error here.
  x.list(); // Even though we can't handle this because we don't know who x is this wasn't an error.
}
```
This seemed to me more of a workaround than what we really want. The reason is that in the code above assigning to `x` is the real problem, and referencing `b` without any method isn't strictly a problem because we don't need to set up any permissions when no method is being used.
Now the approach is:
```wing
let b = new cloud.Bucket();
test "test" {
  let x = b; // This is fine, but when you'll try to use x you'll get an error since we don't do data flow analysis
  x.list(); // I don't know who `x` is so we report an error
}
```
I actually **had** to make this change because of this scenario:
```wing
let a = [new cloud.Bucket()];
test "test" {
  a.at(0).list(); // `a.at(0)` is lifted, and qualified with `list()` - all's great!
  let i = 0;
  a.at(i).list(); // new: `a` is lifted, `at(i)` is evaluated inflight, `list()` is a qualification on an `Bucket` but we don't know which because it's an inflight expression.
}
```
In the `at(i)` example above, without the new error message we simply run fine on the sim and run into a permission error on AWS. I needed the more explicit and correct logic of "you're trying to access an inflight expression that evals to a preflight type which we can't trace to the origin".

> Is it basically that it's ok to lift a bucket of unknown origin as long as you're only using its preflight fields (not calling inflight methods, or eventually getter functions perhaps) on it?

Not really. If it's on unknown origin, then you're not lifting you're referencing a lift:
```wing
// inflight
let x = this.bucket; // `this.bucket` is lifted. `x` is an inflight variable referencing a lift (a client).
log(x.node.id); // here we're trying to access preflight members on a lift (a client). This should be an error!
```
I added another fix to this PR to be explicit about the error above, even though it doesn't compile anyway because it fails JSification. `x.node.id` is considered a preflight expression because `node.id` are preflight fields. So we try to lift the expression and fail because it contains a sub-expression that's inflight `x`. This produces this error:

`Cannot reference an inflight value from within a preflight expression`

I now added a more explicit check for this scenario which fails earlier during type checking:

`Can't access preflight member "node" on inflight instance of type "T"`

Reply:
```
test "test" {
  a.at(0).list(); // `a.at(0)` is lifted, and qualified with `list()` - all's great!
  let i = 0;
  a.at(i).list(); // new: `a` is lifted, `at(i)` is evaluated inflight, `list()` is a qualification on an `Bucket` but we don't know which because it's an inflight expression.
}
```

That the compiler can now distinguish these two kinds of expressions is really impressive

Reply:
> That the compiler can now distinguish these two kinds of expressions is really impressive

I was wracking my brain to get this right. I hope I did. The gist of it is that `x.method(args)` meas that when `method` is executed it needs access to x **and** args. So if either is an inflight expression `method` must be executed inflight. Otherwise we can go either way. 

Then I needed to strike some balance that makes sense when "going either way". At some point I had the following **inflight** code JSify to a no-op: `assert(some_lifted_str.length == 1);` Because the assert itself is phase independent and the args aren't inflight we can lift the entire `assert` (void) call (But that seemed to extreme, the assertion is evaluated preflight during lifting!).